src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { getArticleById } from "../utils/api";
+import Error from './Error';
+
+export default function SingleArticle() {
+    const { article_id } = useParams()
+    const [article, setArticle] = useState(null)
+    const [isLoading, setIsLoading] = useState(true)
+    const [error, setError] = useState(null)
+
+    useEffect(() => {
+        setIsLoading(true);
+        getArticleById(article_id)
+            .then((articleData) => {
+                setArticle(articleData)
+                setIsLoading(false)
+            })
+            .catch((err) => {
+                setError(err)
+                setIsLoading(false)
+            });
+    }, [article_id]);
+
+    if (isLoading) {
+        return <p>Loading article...</p>
+    }
+
+    if (error) {
+        return <Error error={error}/>
+    }
+
+    if(!article) {
```

Comment:
this functionality will not work because of the functionality that you have in your api calls. In the catch block of the api calls you are returning an object so if not article is not found you still return an object so article will be set to the error object. I would remove the catch block in the api calls and just catch the errors in your useEffect that way you can setStates there.